### PR TITLE
New and Improved diagnostic routines for wrf_hydro

### PR DESCRIPTION
**Pull request by mgharamti**
## Description:

Improved DART diagnostic routines for wrf_hydro 

### Fixes issue
 NA

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?

- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.

### Tests
I ran this new diagnostic functionality on a number of hydroDART cases on my glade/scratch

## Checklist for merging

- [x] Updated changelog entry
- [ ] Documentation updated
- [x] Version tag 

## Testing Datasets

- [x] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [ ] No dataset needed

### CODE REVIEW

File: models/wrf_hydro/matlab/plot_connections.m [hunk 1]
```diff
@@ -1,8 +1,8 @@
-function plot_connections(x, ROUTE_FILE, fig_pos, show_gauges_legend, colorbar_title)
+function plot_connections(x, ROUTE_FILE, fig_pos, colorbar_title, tiny_flow)
 
 %% DART software - Copyright UCAR. This open source software is provided
 % by UCAR, "as is", without charge, subject to all terms of use at
-% http://www.image.ucar.edu/DAReS/DART/DART_download
+% https://dart.ucar.edu/software/
```
Comment (hkershaw-brown) [inline models/wrf_hydro/matlab/plot_connections.m:5]:
```suggestion
% http://www.image.ucar.edu/DAReS/DART/DART_download
```

The terms of use are at http://www.image.ucar.edu/DAReS/DART/DART_download

Comment (mgharamti) [reply models/wrf_hydro/matlab/plot_connections.m]: Done

**Review by hkershaw-brown — APPROVED** [review]
This looks good Moha.

I put a suggestion to fix a typo in a comment. And a note about adding a comment to describe whatever dimension 14 is.

File: models/wrf_hydro/matlab/plot_connections.m [hunk 2]
```diff
@@ -11,17 +11,20 @@ function plot_connections(x, ROUTE_FILE, fig_pos, show_gauges_legend, colorbar_t
 %
 % ** fig_pos: Position of the figure; could be separate or a subplot
 %
-% ** show_gauges_legend: Option to display the legend of the gauges
-%
 % ** colorbar_title: title of the colorbar (usuallu units "cms", "m", ...)
```
[the remark this answers is not in the document]
Comment (mgharamti) [reply models/wrf_hydro/matlab/plot_connections.m]: Done

File: models/wrf_hydro/matlab/HydroDARTdiags.m [hunk 3]
```diff
@@ -0,0 +1,816 @@
...
+Nt = Nt_tmp(1);
+
+ncid    = netcdf.open(nc(1).state_mean_pr, 'NC_NOWRITE');
+[~, Nl] = netcdf.inqDim(ncid, 1); % # of links in the domain             
+netcdf.close(ncid);
+
+exp = struct;
+for e = 1:num_exps
+    ncid        = netcdf.open(nc(e).obs_diag, 'NC_NOWRITE');
+    [~, rbins]  = netcdf.inqDim(ncid, 14);
```
[the remark this answers is not in the document]
Comment (mgharamti) [reply models/wrf_hydro/matlab/HydroDARTdiags.m]: Done

**Pull request merged**